docs/datatable-tutorial/add_interactivity.md

```diff
@@ -0,0 +1,241 @@
+```python exec
+import reflex as rx
+from pcweb.base_state import State
+from pcweb.templates.docpage import docdemobox
+from pcweb.pages.docs.datatable_tutorial.states import DataTableState, DataTableState2
+
+```
+
+
+
+
+
+## Adding Interactivity to our DataTable
+
+
+Now we will add interactivity to our datatable. We do this using event handlers and event triggers.
+
+
+For our first interactivity example we will use the `on_cell_clicked` event trigger, this is called when the user clicks on a cell of the data editor. The event trigger receives the coordinates of the cell.
+
+```python
+class DataTableState(rx.State):
+    clicked_cell: str = "Cell clicked: "
+
+    ...
+
+
+    def get_clicked_data(self, pos) -> str:
+        self.clicked_cell = f"Cell clicked: \{pos}"
+        
+```
+
+Here we define a state var called `clicked_cell` that we will use to keep track of the clicked cell. We define an event handler `get_clicked_data` that updates the value of the `clicked_cell` var when it is called. In essence, we have clicked on a cell, called the `on_cell_clicked` event trigger which calls the `get_clicked_data` event handler, which updates the `clicked_cell` var.
+
+
+```python demo
+rx.heading(DataTableState.clicked_cell)
+```
+
+```python demo
+rx.data_editor(
+    columns=DataTableState.cols,
+    data=DataTableState.data,
+    on_cell_clicked=DataTableState.get_clicked_data,
+)
+```
+
+
+The event handler `on_cell_context_menu` can be used in the same way as `on_cell_clicked`, except here the event trigger is called when the user right clicks, i.e. when the cell should show a context menu.
```

Comment:
Do we have any example anywhere of displaying a custom context menu? I feel like this is a bit of a tease if we're not going to show the code. And it's a bit more complicated than it sounds here.

At the end of the day, we might leave this line out, or save it for later in the tutorial/API reference.

Reply:
We give an example of on_group_header_context_menu later on. These are not really custom context menus, it is just an event handler that picks up a right mouse click